OneApp.Business/Services/ConfigurationService.cs

```diff
@@ -1,16 +1,43 @@
 ﻿using AutoMapper;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.Extensions.Logging;
 using OneApp.Business.DTOs;
 using OneApp.Business.Interfaces;
+using OneApp.Contracts.v1;
 using OneApp.Data.Context;
+using OneApp.Data.Models;
+using System.Data;
 
 namespace OneApp.Business.Services;
 
-public class ConfigurationService(DataContext _context, IMapper _mapper) : IConfigurationService
+public class ConfigurationService(
+    DataContext _context,
+    IMapper _mapper,
+    ILogger<ConfigurationService> _logger) : IConfigurationService
 {
-    public Task CreateTenant()
+    #region Public methods
+    public async Task<Guid> CreateTenant(CreateTenantRequest request)
     {
-        throw new NotImplementedException();
+        _logger.LogInformation("Create tenant started.");
+
+        using var transaction = _context.Database.BeginTransaction(IsolationLevel.ReadCommitted).GetDbTransaction();
+        try
+        {
+            _logger.LogInformation($"{nameof(CreateTenant)} transaction scope started.");
+            var tenant = new Tenant { Id = Guid.NewGuid(), Name = request.Name };
```

Comment:
Instead of creating Guid here, can this be auto created and returned with addAsync?